announcement/announcement-tool/tool/src/test/org/sakaiproject/announcement/tool/WebContextSetupTest.java

```diff
@@ -0,0 +1,184 @@
+package org.sakaiproject.announcement.tool;
+
+import lombok.Getter;
+import org.hamcrest.CoreMatchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.sakaiproject.announcement.api.AnnouncementChannel;
+import org.sakaiproject.announcement.api.AnnouncementChannelEdit;
+import org.sakaiproject.announcement.api.AnnouncementService;
+import org.sakaiproject.event.api.UsageSessionService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.portal.util.ToolPortalServlet;
+import org.sakaiproject.test.ModiWebTest;
+import org.sakaiproject.tool.api.ActiveToolManager;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.util.RequestFilter;
+import org.sakaiproject.vm.VelocityServlet;
+import org.springframework.mock.web.*;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.ContextHierarchy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import javax.inject.Inject;
+import javax.servlet.*;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration("src/webapp")
+// With the bullhorn handlers no longer being scanned/created in the kernel, we can load the announcement service in
+// a separate context. For that to work, we must set the global context to ours, so the component manager can find our
+// beans. Or we can customize the kernel context through the hierarchy. This gives us support for disentangling tools
+// from the global state. Announcements is an interesting example because the tool uses its own cover, and the cover
+// caches what it gets back from the component manager... So, because the cover is used in other places during init,
+// we would have to have the new context in place during its own refresh. For now, it's easier to load it in the same
+// context as the kernel, deal with the cover issues later.
```

Comment:
Double check this and remove comment block. Ideally, we can use the hierarchy now, where we don't have to amend the kernel context. That should happen automatically by extending the base class and using `@ContextConfiguration` with no name specified (or a different name).